src/lib.rs

```diff
@@ -124,11 +124,13 @@ pub enum Error {
     Fragmented,
     /// An incoming packet was recognized but was self-contradictory.
     /// E.g. a TCP packet with both SYN and FIN flags set.
+    /// Or a IGMP packet querying address that is not multicast
     Malformed,
     /// An incoming packet was recognized but contradicted internal state.
     /// E.g. a TCP packet addressed to a socket that doesn't exist.
     Dropped,
-
+	/// An unspecified IO error on the raw socket
+	IOError,
```

Comment:
This is completely unrelated to IGMP, and should not exist in its current form anyhow.